platform/platform.go

```diff
@@ -328,16 +328,17 @@ func CheckMachine(ctx context.Context, m Machine) error {
 		}
 		out, stderr, err := m.SSH("systemctl is-system-running")
 		if !bytes.Contains([]byte("initializing starting running stopping"), out) {
-			return nil // stop retrying if the system went haywire
+			return nil // stop retrying if the system went haywire, e.g., "degraded"
 		}
+		// For "running" the exit code is 0 thus err is nil but not for, e.g., "starting" where the exit code is 1
 		if err != nil {
-			return fmt.Errorf("could not check if machine is running: %s: %v: %s", out, err, stderr)
+			return fmt.Errorf("checking if machine is running: systemctl is-system-running: \"%s\". %v: \"%s\"", out, err, stderr)
```

Comment:
Tip: %q already quotes the string.
I'm not sure this message is clear enough, given how many strings are being concatenated... I think we need to be more verbose about what each string is:

Failure while checking if machine is running: systemctl is-system-running returned stdout: %q, stderr: %q, err: %v

Reply:
I thought about `%q` but it also turns newlines to `\n`.

The message can also be about SSH connection errors.

Reply:
Sorry, I guess the string should be:
failure checking if machine is running: systemctl is-system-running returned stdout: %q, stderr: %q, err: %v

Reply:
Thanks, I also tried to include the output of `systemctl list-jobs` now if `starting` is the last state.